bundle-core/src/main/java/net/discdd/utils/BundleUtils.java

```diff
@@ -337,7 +341,11 @@ public static void encryptPayloadAndCreateBundle(Encrypter payloadEncryptor, ECP
         outerJar.createEntry(SecurityUtils.BUNDLEID_FILENAME, encryptedBundleId.getBytes());
 
         // store the keys
-        outerJar.createEntry(SecurityUtils.CLIENT_IDENTITY_KEY, createEncodedPublicKeyBytes(clientIdentityPublicKey));
+        try {
+            outerJar.createEntry(SecurityUtils.CLIENT_IDENTITY_KEY, createEncryptedEncodedPublicKeyBytes(clientIdentityPublicKey, serverIdentityPublicKey));
+        } catch (GeneralSecurityException e) {
+            throw new RuntimeException(e);
```

Comment:
IOException would be better: throw new IOException("General Security Exception", e);